### PR TITLE
docs: Fix Environment Variables docs

### PR DESCRIPTION
**Pull request by elliotwaite**
In addition to the changes listed in this PR, I noticed `EDGEDB_SERVER_DEFAULT_AUTH_METHOD` is listed twice on the Environment Variables page, once [here](https://www.edgedb.com/docs/reference/environment#edgedb-server-default-auth-method) and again [here](https://www.edgedb.com/docs/reference/environment#id1). It seems like the description of the first one is actually for a different env var since it mentions that it's deprecated, but I couldn't figure out which env var it was supposed to be.

### CODE REVIEW

**Comment by Dhghomon** [comment]
@elliotwaite Interesting puzzle. Fortunately the env vars on that page seem to be in alphabetical order which leads me to believe that it might be EDGEDB_SERVER_ALLOW_INSECURE_HTTP_CLIENTS which @raddevon once asked about documenting. Devon, do you think that's the deprecated env var that fits in here?

**Comment by elliotwaite** [comment]
Good point about the alphabetical order. From the current codebase, I found these env vars that could fit between `EDGEDB_SERVER_ADMIN_UI` and `EDGEDB_SERVER_BACKEND_DSN`:
* `EDGEDB_SERVER_ALLOW_INSECURE_BINARY_CLIENTS`
* `EDGEDB_SERVER_ALLOW_INSECURE_HTTP_CLIENTS`
* `EDGEDB_SERVER_AUTH_SMTP_CONCURRENCY`
* `EDGEDB_SERVER_BACKEND_ADAPTIVE_HA_DISCONNECT_PERCENT`
* `EDGEDB_SERVER_BACKEND_ADAPTIVE_HA_UNHEALTHY_MIN_TIME`
* `EDGEDB_SERVER_BACKEND_CAPABILITIES`

...but none of those seem to fit the description:
> Optionally specifies the authentication method used by the server instance. Supported values are SCRAM (the default) and Trust. When set to Trust, the database will allow complete unauthenticated access for all who have access to the database port.

**Comment by elliotwaite** [comment]
I just updated the PR to replace that block with two blocks for both:
* `EDGEDB_SERVER_ALLOW_INSECURE_BINARY_CLIENTS`
* `EDGEDB_SERVER_ALLOW_INSECURE_HTTP_CLIENTS`

Let me know what you think.

**Comment by Dhghomon** [comment]
> Good point about the alphabetical order. From the current codebase, I found these env vars that could fit between `EDGEDB_SERVER_ADMIN_UI` and `EDGEDB_SERVER_BACKEND_DSN`:
> 
> * `EDGEDB_SERVER_ALLOW_INSECURE_BINARY_CLIENTS`
> * `EDGEDB_SERVER_ALLOW_INSECURE_HTTP_CLIENTS`
> * `EDGEDB_SERVER_AUTH_SMTP_CONCURRENCY`
> * `EDGEDB_SERVER_BACKEND_ADAPTIVE_HA_DISCONNECT_PERCENT`
> * `EDGEDB_SERVER_BACKEND_ADAPTIVE_HA_UNHEALTHY_MIN_TIME`
> * `EDGEDB_SERVER_BACKEND_CAPABILITIES`
> 
> ...but none of those seem to fit the description:
> 
> > Optionally specifies the authentication method used by the server instance. Supported values are SCRAM (the default) and Trust. When set to Trust, the database will allow complete unauthenticated access for all who have access to the database port.
> 
> But maybe the description was also incorrectly overwritten. In which case `EDGEDB_SERVER_ALLOW_INSECURE_BINARY_CLIENTS ` or `EDGEDB_SERVER_ALLOW_INSECURE_HTTP_CLIENTS` would seem to make the most sense. And both of those [were deprecated in 1.0 RC 2](https://github.com/edgedb/edgedb/blob/3965431b242ce4c37e863a9371ceb2c07fa3982d/docs/changelog/1_0_rc2.rst#environment-variable-renames), being renamed `EDGEDB_SERVER_BINARY_ENDPOINT_SECURITY ` and `EDGEDB_SERVER_HTTP_ENDPOINT_SECURITY ` respectively.

Yeah, I think that's the one! Looks good to me, will approve and see if Devon has any comments before merging.

**Review by Dhghomon — APPROVED** [review]

**Comment by raddevon** [comment]
Looks great. Thank you for catching this and submitting a fix, @elliotwaite!

**Pull request merged**
